Add let else statements (#165)
rust-lang/fmt-rfcs: guide/statements.md

    foo();
}
```

### Let else statements

> The `else` should have once space before and after. It can also be at the beginning of a line with a space after.

Reviewer: ```suggestion
The keyword `else` should either appear one space after the expression, or on a subsequent line with indentation matching the keyword `let`. There should always be a space between `else` and `{`.
```

> If the `else` block only contains only an expression, the entire statement should be on one line if it fits.

Reviewer: ```suggestion
If the `else` block only contains only a single expression (e.g. `return x` or `continue` or `break x`), the entire statement should appear on one line if it fits.
```

```rust
let Some(1) = opt else { return };
```

> If the line needs to be broken, prefer to break after the opening brace of the `else` block

Reviewer: ```suggestion
If the line needs to be broken, prefer to break after the opening brace of the `else` block first, before inserting any line breaks into the initializer or pattern:
```

```rust
let Some(1) = opt else {
    return
};
```

If the `else` and opening brace do not fit on the same line as the initializer,

> break before `else`. If a line begins with `else`, it should be indented at the same level as `let`,

Reviewer: ```suggestion
break before `else`. If a line begins with `else` (potentially preceded by closing braces or parentheses or similar), it should be indented at the same level as the matching `let`,
```

> and the block should be on the same line if it fits.

Reviewer: ```suggestion
and there should always be a newline after the opening brace so that the block always appears on a subsequent line. This calls attention to the `else` block attached to the `let`, making it easy to visually distinguish a `let`-`else` from a plain `let`.
```

Author: > This calls attention to the `else` block

The style guide does not need to say "why" right?

I think what you are suggesting here should not be a dependent clause of "if a line begins with `else`". How about this?

> If the `else` is not on the same line as `let`, the `else` block should not be written in one line.

I'm assuming that we can lean on the "blocks" section of the guide and so "not written in one line" is specific enough.

Regardless of how it's worded, I agree, this is more consistent with if/else.

```rust
let MyStruct { foo } = ({
    statement;
    fun()

> }) else { return };

Reviewer: ```suggestion
}) else {
    return
};
```

let Some(1) = opt

> else { return };

Reviewer: ```suggestion
else {
    return
};
```

let Some(1) = opt
else {
    println!("nope");
    return;
};
```

If the last line of the initializer expression is indented past `let`,
the `else` should be broken to the next line.

```rust
let Foo { bar } = foo
    .method()
else {

>     return;

Reviewer: ```suggestion
    return
```

Consistency with other examples.

};

let MyStruct { foo: Some(1) } =
    some_variable
else {

>     return;

Reviewer: ```suggestion
    return
```

Consistency with other examples
